server/server/src/file_service.rs

```diff
@@ -41,22 +45,41 @@ pub async fn upsert_file_metadata(
         )?
         .to_lazy();
 
+        let old_usage = get_usage(&tree, db.sizes.data(), None)
+            .unwrap_or_default()
+            .iter()
+            .map(|f| f.size_bytes)
+            .sum::<u64>();
+
         for id in tree.owned_ids() {
             if tree.calculate_deleted(&id)? {
                 prior_deleted.insert(id);
             }
         }
 
         let mut tree = tree.stage_diff(request.updates.clone())?;
-        tree.validate(req_owner)?;
-        let mut tree = tree.promote()?;
-
         for id in tree.owned_ids() {
             if tree.calculate_deleted(&id)? {
                 current_deleted.insert(id);
             }
         }
 
+        tree.validate(req_owner)?;
+
+        let new_usage = get_usage(&tree, db.sizes.data(), Some(&current_deleted))
```

Comment:
yeah so this tree captures the files that are hypothetically going to be deleted if allowed (because it's a staged tree).

and `sizes` already knows what files are previously deleted (because it will have no sizes for those files).

So you can probably get rid of the extra optional argument (`Some(&current_deleted))`).

lmk if you think I've missed something.